### PR TITLE
update awslambda to use _X_AMZN_TRACE_ID as a Span Link

### PR DESCRIPTION
**Pull request by tsloughter**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Ran `test-instrumentation-aws-lambda` environment tests, adding `test_links_from_lambda_event` and updating `test_parent_context_from_lambda_event`

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [X] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [X] Followed the style guidelines of this project
- [X] Changelogs have been updated
- [X] Unit tests have been added
- [X] Documentation has been updated

### CODE REVIEW

**Comment by tsloughter** [comment]
Forgot to mention: This is to make the Lambda instrumentation line up with an update to the spec that should be merged soon https://github.com/open-telemetry/opentelemetry-specification/pull/3166

File: instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py [hunk 1]
```diff
@@ -166,28 +168,45 @@ def _determine_parent_context(
...
+) -> Optional[Sequence[Link]]:
+    """Determine if a Link should be added to the Span based on the
+    environment variable `_X_AMZN_TRACE_ID`.
+
+
+    See more:
+    https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-lambda.md
+
+    Args:
+        disable_aws_context_propagation: By default, this instrumentation
```
Comment (codeboten) [inline instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py:190]: Is this mechanism still necessary?

Comment (tsloughter) [reply instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py:190]: Nope. Forgot we decided to remove those.

Comment (tsloughter) [reply instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py]: Removed.

File: instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py [hunk 2]
```diff
@@ -190,6 +168,34 @@ def _determine_parent_context(
     return parent_context
 
 
+def _determine_links() -> Optional[Sequence[Link]]:
+    """Determine if a Link should be added to the Span based on the
+    environment variable `_X_AMZN_TRACE_ID`.
+
+
+    See more:
+    https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-lambda.md
```
Comment (ronyis) [inline instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py:177]:
```suggestion
    https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-lambda.md#aws-x-ray-environment-span-link
```

File: instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py [hunk 3]
```diff
@@ -190,6 +168,34 @@ def _determine_parent_context(
...
+
+    Returns:
+        A Link or None
+    """
+    env_context = None
```
Comment (ronyis) [inline instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py:182]:
```suggestion
```

File: instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py [hunk 4]
```diff
@@ -190,6 +168,33 @@ def _determine_parent_context(
...
+    xray_env_var = os.environ.get(_X_AMZN_TRACE_ID)
+
+    if xray_env_var:
+        env_context = AwsXRayPropagator().extract(
+            {TRACE_HEADER_KEY: xray_env_var}
+        )
+
+        span_context = get_current_span(env_context).get_span_context()
+        if span_context.is_valid:
+            links = [Link(span_context)]
```
Comment (martinkuba) [inline instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py:193]: The link should have a `source=x-ray-env` attribute per the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/7c2312534b539fc599fd1f0901e874c6e489c37a/specification/trace/semantic_conventions/instrumentation/aws-lambda.md#aws-x-ray-environment-span-link).

**Comment by tsloughter** [comment]
yup, thanks @martinkuba I've updated the code to include the attribute

**Comment by srikanthccv** [comment]
@codeboten @martinkuba @ronyis PTAL

**Comment by shuwpan** [comment]
Hi team, I tried using the layer built from this PR, I do get the spanlink, but the value is not what I am expecting. I might be missing something, according to the spec, the spanlink should contain the producer's traceId out of the box, but it was not there. I knew the _X_AMZN_TRACE_ID is from AMZ, not sure whether is it configurable from outside. So just wanted to start the conversation. Please redirect me to the proper place if this is not. I have all the logs and setups in this stackoverflow post. https://stackoverflow.com/questions/76125310/aws-lambda-sqs-opentelemetry-python-link-field-value-can-i-configure-it-to-be

**Review by ocelotl — APPROVED** [review]

**Pull request merged**